src/lib/userAuthCall/transferFrom/index.ts

```diff
@@ -0,0 +1,141 @@
+  UserAuthCallWithTransferFromPayloadToSign,
+  UserAuthCallWithTransferFromRequest,
+  UserAuthCallWithTransferFromRequestOptionalParameters,
+  UserAuthCallWithTransferFromStruct,
+} from "./types";
+
+const getPayloadToSign = (
+  struct: UserAuthCallWithTransferFromStruct
+): UserAuthCallWithTransferFromPayloadToSign => {
+  const verifyingContract = getRelayAddress(struct.chainId as number);
+  const domain = getEIP712Domain(
+    "GelatoRelayWithTransferFrom",
+    "1",
+    struct.chainId as number,
+    verifyingContract
+  );
+  return {
+    domain,
+    types: EIP712_USER_AUTH_CALL_WITH_TRANSFER_FROM_TYPE_DATA,
+    primaryType: "UserAuthCallWithTransferFrom",
+    message: struct,
+  };
+};
+
+const mapRequestToStruct = (
+  request: UserAuthCallWithTransferFromRequest,
+  override: Partial<UserAuthCallWithTransferFromRequestOptionalParameters>
+): UserAuthCallWithTransferFromStruct => {
+  if (!override.userNonce && !request.userNonce) {
+    throw new Error(`userNonce is not found in the request, nor fetched`);
+  }
+  if (!override.userDeadline && !request.userDeadline) {
+    throw new Error(`userDeadline is not found in the request, nor fetched`);
+  }
+  return {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    userNonce: override.userNonce ?? request.userNonce!,
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    userDeadline: override.userDeadline ?? request.userDeadline!,
+    chainId: request.chainId,
+    target: getAddress(request.target as string),
+    data: request.data,
+    user: getAddress(request.user as string),
+    paymentType: PaymentType.TransferFrom,
+    feeToken: getAddress(request.feeToken as string),
+    maxFee: request.maxFee,
+  };
+};
+
+const post = async (
+  request: UserAuthCallWithTransferFromStruct &
+    RelayRequestOptions &
+    UserAuthSignature
+): Promise<RelayResponse> => {
+  try {
+    const response = await axios.post(
+      `${GELATO_RELAY_URL}/relays/v2/user-auth-call`,
+      request
+    );
+    return response.data;
+  } catch (error) {
+    const errorMessage =
+      error.response?.data?.message ??
+      error.message ??
+      DEFAULT_INTERNAL_ERROR_MESSAGE;
+    throw new Error(errorMessage);
+  }
+};
+
+const populateOptionalParameters = async (
+  request: UserAuthCallWithTransferFromRequest,
+  provider: providers.Web3Provider
+): Promise<Partial<UserAuthCallWithTransferFromRequestOptionalParameters>> => {
+  const parametersToOverride: Partial<UserAuthCallWithTransferFromRequestOptionalParameters> =
+    {};
+  if (!request.userDeadline) {
+    parametersToOverride.userDeadline = calculateDeadline(DEFAULT_DEADLINE_GAP);
+  }
+  if (!request.userNonce) {
+    parametersToOverride.userNonce = (
+      (await getUserNonce(
+        request.chainId as number,
+        request.user as string,
+        provider
+      )) as BigNumber
+    ).toNumber();
+  }
+
+  return parametersToOverride;
+};
```

Comment:
this method is duplicated as well, it should be an internal/util method, for example `_getUserOptionalParams`